azext_iot/operations/hub.py

```diff
@@ -1940,6 +1950,45 @@ def iot_get_sas_token(
     }
 
 
+def _iot_build_sas_token_from_cs(connection_string, duration=3600):
+    uri = None
+    policy = None
+    key = None
+
+    parsed_cs = None
+    all_parsers = [
+        ConnectionStringParser.Module,
+        ConnectionStringParser.Device,
+        ConnectionStringParser.PnP,
+        ConnectionStringParser.IotHub,
+    ]
+
+    for parser in all_parsers:
+        try:
+            parsed_cs = parser(connection_string)
+
+            if "SharedAccessKeyName" in parsed_cs:
+                policy = parsed_cs["SharedAccessKeyName"]
+            key = parsed_cs["SharedAccessKey"]
+
+            if parser in [ConnectionStringParser.IotHub, ConnectionStringParser.PnP]:
+                uri = parsed_cs["HostName"]
+            elif parser == ConnectionStringParser.Module:
+                uri = "{}/devices/{}/modules/{}".format(
+                    parsed_cs["HostName"], parsed_cs["DeviceId"], parsed_cs["ModuleId"]
+                )
+            elif parser == ConnectionStringParser.Device:
+                uri = "{}/devices/{}".format(parsed_cs["HostName"], parsed_cs["DeviceId"])
+            else:
+                raise CLIError("Connection String did not match any presets")
+
+            return SasTokenAuthentication(uri, policy, key, duration)
+        except ValueError:
+            continue
+
+    raise CLIError("Connection String did not match any presets")
```

Comment:
I think we could make this error mode have a bit clearer message ...maybe some tweaks in wording around the given connection string value not being in a supported form. Or `Connection String was not in a supported format.` -- `presets` is throwing me off.

Reply:
Can we change the wording in the this CLIError as well.